src/api/nes/nes.client.ts

```diff
@@ -30,78 +35,112 @@ export class NesApolloClient implements NesClient {
   try {
     const batches = createBatches(purls, batchSize);
     debugLogger('Processing %d batches', batches.length);
 
-    const results = await Promise.allSettled(
-      batches.map((batch, index) => {
-        debugLogger('Starting batch %d', index + 1);
-        return submitScan(batch);
-      }),
-    );
-
-    const successfulResults: ScanResult[] = [];
-    const errors: string[] = [];
-
-    for (const [index, result] of results.entries()) {
-      if (result.status === 'fulfilled') {
-        debugLogger('Batch %d completed successfully', index + 1);
-        successfulResults.push(result.value);
-      } else {
-        debugLogger('Batch %d failed: %s', index + 1, result.reason);
-        errors.push(`Batch ${index + 1}: ${result.reason}`);
-      }
+    if (batches.length < 0) {
+      throw new Error('No batches to process');
     }
-
-    if (successfulResults.length === 0) {
-      throw new Error(`All batches failed:\n${errors.join('\n')}`);
+    if (batches.length === 0) {
+      return {
+        components: new Map<string, InsightsEolScanComponent>(),
+        message: 'No batches to process',
+        success: true,
+        warnings: [],
+      } satisfies ScanResult;
     }
 
-    const combinedResults = combineScanResults(successfulResults);
-    if (errors.length > 0) {
-      combinedResults.success = false;
-      combinedResults.message = `Errors encountered:\n${errors.join('\n')}`;
-    }
-
-    return combinedResults;
+    const results = await processBatches(batches, options);
+    return handleBatchResults(results);
   } catch (error) {
     debugLogger('Fatal error in batchSubmitPurls: %s', error);
     throw new Error(`Failed to process purls: ${error instanceof Error ? error.message : String(error)}`);
   }
-}
+};
+
+export const createBatches = (items: string[], batchSize: number): string[][] =>
+  Array.from({ length: Math.ceil(items.length / batchSize) }, (_, i) =>
+    items.slice(i * batchSize, (i + 1) * batchSize),
+  );
+
+export const processBatch = async ({
+  batch,
+  index,
+  totalPages,
+  scanOptions,
+  previousScanId,
+}: ProcessBatchOptions): Promise<InsightsEolScanResult> => {
+  const page = index + 1;
+  if (page > totalPages) {
+    throw new Error('Total pages exceeded');
+  }
+
+  debugLogger('Processing batch %d of %d', page, totalPages);
+  const result = await submitScan(batch, {
+    ...scanOptions,
+    page,
+    totalPages,
+    scanId: previousScanId,
+  });
+  return result;
+};
+
+export const processBatches = async (
+  batches: string[][],
+  scanOptions: ScanInputOptions,
+): Promise<InsightsEolScanResult[]> => {
+  const totalPages = batches.length;
+  const results: InsightsEolScanResult[] = [];
+
+  for (const [index, batch] of batches.entries()) {
+    const previousScanId = index > 0 ? results[index - 1].scanId : undefined;
+    const result = await processBatch({
+      batch,
+      index,
+      totalPages,
+      scanOptions,
+      previousScanId,
```

Comment:
```suggestion
    const result = await processBatch({
      batch,
      index,
      totalPages,
      scanOptions,
      previousScanId: results[index - 1]?.scanId,
```

Accessing a negative index simply returns undefined so it can be simplifies

Reply:
Done [0367622](https://github.com/herodevs/cli/pull/149/commits/0367622515243db01b0ae72edf0325f93de6a317)